resolwe/flow/models/utils/duplicate.py

```diff
@@ -240,13 +240,13 @@ def processor(datum):
 
         """
         if inherit_collection:
-            if datum.collection.id in collection_mapping:
+            if datum.collection and datum.collection.id in collection_mapping:
                 datum.collection = collection_mapping[datum.collection.pk]
         else:
             datum.collection = None
 
         if inherit_entity:
```

Comment:
Maybe you could just rewrite the first condition to `if inherit_entity and entity.collection`?